### PR TITLE
play.nteract.io and a bunch of @nteract/messaging improvements

### PR DESCRIPTION
**Pull request by rgbkrk**
screen shot 2017-12-01 at 11 17 02 pm

This hooks up the little play app to properly execute code and use the new next gen nteract components (for styling mostly).

A few things I've noticed that need some reform:

* The `WebSocketSubject` requires that you `JSON.stringify` the payload, which makes sense given the WebSocket abstraction. Our version of the subject should probably:
  * `JSON.stringify` messages for you
  * Set the session ID and username
  * Set the appropriate channel for you based on the message type (@captainsafia mentioned this a couple weeks ago)

TODO:

* [x] Clean up messaging and use `@nteract/messaging`
* [x] Introduce automatic session and channelization to messaging
* [x] Bind ~~shift-enter~~ / ctrl-enter / cmd-enter
* [x] Move play button into header (?)
* [x] Merge headers together
* [x] Clicking "below" the editor should focus the editor

Future enhancements to make this a fully useful tool:

* Ability to save these "snippets" / "fiddles"
* Ability to "fork" a fiddle and keep provenance
* A way to specify the repo & ref

### CODE REVIEW

**Comment by rgbkrk** [comment]
Very cool extremely neat side effects of this work:

* An application using a channel no longer needs sessions. This has been a thorn in the desktop app implementation for a long time. I'm going to follow up with session cleanup there too. This will make the nbextension / web app way cleaner
* Multiplexed channels are much easier to work with

File: packages/messaging/src/messages.js [hunk 1]
```diff
@@ -10,17 +10,65 @@ import * as uuid from "uuid";
...
   content?: Object = {}
 ): JupyterMessage<*, *> {
+  const channel = whichChannel(header.msg_type);
+
   return {
     header: {
       msg_id: uuid.v4(),
-      date: new Date(),
-      version: "5.1",
+      date: new Date().toISOString(),
```
Comment (rgbkrk) [inline packages/messaging/src/messages.js:67]: node's `new Date()` serializes to an ISOString whereas browser implementations can differ. Explicitly using `toISOString` is much wiser here.

File: packages/editor/src/styles.js [hunk 2]
```diff
@@ -4,6 +4,9 @@ import css from "styled-jsx/css";
 
 export default css`
   /* completions styles */
+  :global(.CodeMirror) {
+    height: 100%;
```
Comment (rgbkrk) [inline packages/editor/src/styles.js:8]: This matches the parent element's height and in desktop is basically a no-op due to other CSS. Here in `play`, this makes it fit the page.

File: packages/messaging/src/messages.js [hunk 3]
```diff
@@ -29,14 +77,16 @@ export function message(
     },
     metadata: {},
     parent_header: {},
-    content
+    content,
+    channel,
```
Comment (rgbkrk) [inline packages/messaging/src/messages.js:81]: This allows our multiplexed websocket subject to pass the message straight on the through. Coming to the desktop / raw ZeroMQ version will be routing based on this.

File: packages/messaging/src/messages.js [hunk 4]
```diff
@@ -266,3 +316,7 @@ export function executeInput(
     content
   );
 }
+
+export function kernelInfoRequest(sessionInfo?: SessionInfo) {
+  return message({ msg_type: "kernel_info_request", ...sessionInfo });
```
Comment (rgbkrk) [inline packages/messaging/src/messages.js:321]: I look forward to nuking all the `sessionInfo`s here due to better `Subject` handling in Rx-land.

**Comment by rgbkrk** [comment]
screen shot 2017-12-01 at 11 16 35 pm

**Comment by rgbkrk** [comment]
screen shot 2017-12-01 at 11 17 02 pm

File: packages/messaging/src/messages.js [hunk 5]
```diff
@@ -29,14 +77,16 @@ export function message(
...
+    channel,
+    buffers: []
   };
 }
 
 /**
  * An execute request creator
  *
- * > executeRequest({ username: 'kyle', session: '123' }, 'print("hey")', { 'allow_stdin': true })
+ * > executeRequest(print("hey")', { 'silent': true })
```
Comment (betatim) [inline packages/messaging/src/messages.js]: `executeRequest('print("hey")', { 'silent': true })` (missing tick before print)

**Comment by betatim** [comment]
~~add key press handler or~~ new PR?

**Comment by betatim** [comment]
Should we merge this then?

**Comment by rgbkrk** [comment]
so ready

**Comment by betatim** [comment]
🚀

**Pull request merged**